### PR TITLE
feat: add e2e-test-plugin

### PR DESCRIPTION
**Pull request by Two-Hearts**
Providing a notation mock plugin for notation Github actions E2E test workflows.

This PR targets the `e2e-test-pluing` branch and is not intended to be merged into Main.

### CODE REVIEW

File: .github/e2e-test/plugin/.gitignore
Comment (shizhMSFT) [inline .github/e2e-test/plugin/.gitignore:1]: I think we can move `.github/e2e-test` to `tests` folder

File: .github/workflows/release-notation-e2e-test-plugin.yml
Comment (shizhMSFT) [inline .github/workflows/release-notation-e2e-test-plugin.yml:1]: We don't need to release the plugin but to build and check in the binaries to the a branch of this repo.

File: README.md [hunk 1]
```diff
@@ -1,117 +1,5 @@
-# Notation Github Actions
-Github Actions for [Notation](https://notaryproject.dev/).
+# Notation Github Actions e2e-test-plugin
+This branch provides a mock plugin for notation Github actions E2E test workflows.
```
Comment (shizhMSFT) [inline README.md:2]: We should set up a branch policy for the `e2e-test-plugin` branch.

File: tests/plugin/go.mod [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module notation-e2e-test-plugin
```
Comment (shizhMSFT) [inline tests/plugin/go.mod:1]:
```suggestion
module github.com/notaryproject/notation-action/tests/plugin
```

File: tests/plugin/cmd/notation-e2e-test-plugin/metadata.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runGetMetadata()
+		},
+	}
+}
+
+func runGetMetadata() error {
+	return json.NewEncoder(os.Stdout).Encode(proto.GetMetadataResponse{
+		Name:                      "e2e-test-plugin",
```
Comment (shizhMSFT) [inline tests/plugin/cmd/notation-e2e-test-plugin/metadata.go:36]:
```suggestion
		Name:                      "notation-e2e-test-plugin",
```

File: tests/plugin/cmd/notation-e2e-test-plugin/metadata.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		Description:               "Sign artifacts with local keys",
```
Comment (shizhMSFT) [inline tests/plugin/cmd/notation-e2e-test-plugin/metadata.go:37]:
```suggestion
		Description:               "Sign artifacts with local keys for testing purposes",
```

File: README.md (hunk 1 above)
Comment (Two-Hearts) [reply README.md:2]: Sounds good. I think we could set it up after this PR gets merged?

File: tests/plugin/cmd/notation-e2e-test-plugin/metadata.go (hunk 3 above)
Comment (Two-Hearts) [reply tests/plugin/cmd/notation-e2e-test-plugin/metadata.go:36]: @shizhMSFT hmm, I'm not sure if we should add the `notation` prefix? For akv plugin, it has `azure-kv` here.

Comment (shizhMSFT) [reply tests/plugin/cmd/notation-e2e-test-plugin/metadata.go:36]: Oh, it is the metadata name. I thought it was the binary name.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**